google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/project/ProjectSelector.java

```diff
@@ -155,7 +155,8 @@ private void updateCloudProjectSelection(CloudProject selection) {
         GoogleLoginIcons.getScaledUserIcon(ACCOUNT_ICON_SIZE, loggedInUser.orElse(null)));
   }
 
-  private void notifyProjectSelectionListeners() {
+  @VisibleForTesting
+  public void notifyProjectSelectionListeners() {
```

Comment:
@ivanporty I really don't want to do this, but it was a quick and dirty way to make things testable. Thinking of a better way.

Reply:
@ivanporty to make this better I'm thinking of the following. let me know your thoughts:

- changing `ProjectSelector#handleOpenProjectSelectionDialog` -> `ProjecSelector#showDialog()` (just to make things clearer as a public method)
- making the above part of the public api of the selector
- reverting the change here (line 159) and using the showDialog method for unit testing

Reply:
It still feels a bit too open - project selector calling into dialog is internal implementation. The only other idea I might have is keep the listener on project selector as a field and provide `@VisibleForTesting` access to it.
or, even simpler, why don't we just directly call `showApiEnablementUi()` instead - that's the method that gets called on project selection change, correct?

Reply:
calling showApiEnablementUi directly wouldn't be much of a test since the test in question is just asserting that the fields set in showApiEnablementUi are actually set. The idea is to test that when a cloud project is correctly set that the panel UI is correctly initialized.

The way I justified exposing showDialog in my head was that I could sort of imagine a UX where we'd want to automatically open up the dialog for the user. But I don't really see that happening.

I could see exposing the listeners instead.